packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -395,6 +437,68 @@ export class ChainAdapter implements IChainAdapter<KnownChainIds.SolanaMainnet>
     return Buffer.from(transaction.serialize()).toString('base64')
   }
 
+  private async buildTokenTransferInstructions({
+    from,
+    to,
+    tokenId,
+    value,
+  }: {
+    from: string
+    to: string
+    tokenId: string
+    value: string
+  }): Promise<TransactionInstruction[]> {
+    const instructions: TransactionInstruction[] = []
+
+    const destinationTokenAccount = getAssociatedTokenAddressSync(
+      new PublicKey(tokenId),
+      new PublicKey(to),
+    )
+
+    // check if destination token account exists and add creation instruction if it doesn't
+    try {
+      await getAccount(this.connection, destinationTokenAccount)
+    } catch (err) {
+      if (
+        err instanceof TokenAccountNotFoundError ||
+        err instanceof TokenInvalidAccountOwnerError
+      ) {
+        instructions.push(
+          createAssociatedTokenAccountInstruction(
+            // sender pays for creation of the token account
+            new PublicKey(from),
+            destinationTokenAccount,
+            new PublicKey(to),
+            new PublicKey(tokenId),
+          ),
+        )
+      }
+    }
```

Comment:
Based on what I am reading, the transaction will either fail if the balance drops below the rent threshold, but not 0, or if completely withdrawn (balance is 0), the account will be automatically garbage collected. With this being the case, an explicit close doesn't seem necessary, however, there may be some edge case scenarios around rent thresholds or create account fees that we will want to take into account.
- warn user when try to send tokens that will result in balance under rent threshold and fail
- take into account create account fee as part of balance checks

Reply:
I have a lot of dust accounts linked with my wallet, we will need to verify on the long term if sending the full amount is triggering a garbage collect because I feel like it's not that magical, not a blocker anyway but something we have to keep in mind